Jenkinsfile

```
@@ -177,6 +177,7 @@ pipeline {
                   expression { return env.GITHUB_COMMENT?.contains('integration tests') }
                   expression { matchesPrLabel(label: 'ci:agent-integration') }
                   expression { return env.CHANGE_ID != null && !pullRequest.draft }
+                  not { changeRequest() }
```

Wouldn't that be equivalent to what you proposed?
```suggestion
                  expression { return env.CHANGE_ID == null || !pullRequest.draft }
```

---

I think so, although it's not purely declarative based, but simplifies two conditions in one. No strong opinion about this

---

I would rather keep those into two distinct conditions for clarity (if you don't mind).